src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,328 @@
+    config = {
+        'cbsdRecords': [{'registrationRequests':[device_1,device_2],'grantRequests':[grant_record_1,grant_record_2]},
+                        {'registrationRequests':[device_3,device_4],'grantRequests':[grant_record_3,grant_record_4]}],
+        'IAPRecords': [[
+      { 'type': 'FSS','Record': [fss_record_1] },
+      { 'type': 'ESC','Record': [esc_record_1] } ],
+      [ { 'type': 'GWPZ','Record': [gwpz_record_1]},
+      { 'type': 'FSS', 'Record': [fss_record_2]}]],
+	'sas_hareness': [{'name': 'SAS-TH-1',
+			'url': ['localhost', '9001'],
+			'cert_file': 'certs/server_a.cert',
+			'key_file': 'certs/server_a.key'},
+	                {
+			'name': 'SAS-TH-2',
+			'url': ['localhost', '9002'],
+			'cert_file': 'certs/server_b.cert',
+			'key_file': 'certs/server_b.key'
+		        }],
+        'sas_uut': {'name': 'SAS-UUT',
+                        'url': ['localhost', '9000'],
+                        'cert_file': 'certs/server.cert',
+                        'key_file': 'certs/server.key'}, 
+        'K':3, 
+        'headroom':{'MgPpa':2,'MgGwpz':2,'MgCochannel':2,'MgBlocking':2,'MgOobe':2,'MgEsc':2},
+        'DPAactiveInfo':[],
+        'DPAdeactiveInfo':[]
+       }
+    
+    writeConfig(filename, config)
+
+
+  @configurable_testcase(generate_MCP_1_default_config) 
+  def test_WINNF_FT_S_MCP_1(self,config_filename):
+    """SAS manages a mix of GAA and PAL Grants in 3550 MHz
+       to 3700 MHz to protect configurable IAP-protected entities and DPAs
+    """
+    config = loadConfig(config_filename)
+    pre_iap_headroom = config ['headroom']
+
+    
+    #Step 1 : Load DPAs
+    self._sas_admin.TriggerLoadDpas()
+
+    #STEP 2 : ESC informs SAS about inactive DPA
+    self._sas_admin.TriggerBulkDpaActivation({'activate':False})
+
+    #Step 3 : creates multiple SAS TH, and Load predefined FAD,CBSD 
+    self.createSASTH(config)
+    self.configureFadForEachSASTH(0)
+    
+    #fad objects get the fad dumps from the remote SAS configured with 'url'
+    fad_object_th = []
+    for N in len(config['sas_hareness']):
+       cert = os.path.join('certs',config['sas_hareness'][N]['cert_file'])
+       key  = os.path.join('certs',config['sas_hareness'][N]['key_file'])
+       fad_object_th.append(FullActivityDump(cert,key,config['sas_hareness'][N]['url']))
+       fad_records.append(fad_object_th[N])
+
+    #for UUT creating fad object
+    cert = os.path.join('certs',config['sas_uut']['cert_file'])
+    key  = os.path.join('certs',config['sas_uut']['key_file'])
+    fad_uut_object = FullActivityDump(cert,key,config['sas_uut']['url'])
+    fad_records.append(fad_uut_object)
+
+    # informing SAS UUT about SAS TH 
+    for N in len(config['sas_hareness']):
+       self._sas_admin.InjectPeerSas({'certificateHash': config[N]['certificateHash'],
+                                     'url': 'https://' + config[N] ['url']+ '/dump'}) 
+    
+
+    #Step 4,5 : Inject IAP protected entities into UUT
+    for k in  config['K']:	
```

Comment:
This does not work since config['K'] is a scalar.

Reply:
updated   with range(config['K'])